src/main/java/projet/projetstage02/service/GestionnaireService.java

```diff
@@ -409,4 +408,139 @@ public ContractsDTO getContractsToEvaluateMillieuStage() {
         });
         return ContractsDTO.builder().contracts(contracts).build();
     }
+
+    public void createEvaluationMillieuStagePDF(long contractId) throws NonExistentEntityException, NonExistentOfferExeption, DocumentException, EmptySignatureException {
+        Optional<EvaluationMillieuStage> optional = evaluationMillieuStageRepository.findByContractId(contractId);
+        if (optional.isEmpty()) {
+            throw new NonExistentEntityException();
+        }
+        EvaluationMillieuStage evaluationMillieuStage = optional.get();
+        MillieuStageEvaluationInfoDTO millieuStageEvaluationInfoDTO =
+                getMillieuEvaluationInfoForContract(contractId);
+
+        evaluationMillieuStagePDFRepository.save(EvaluationPDF.builder()
+                .pdf(PDFCreationUtil.createPDFFromMap("Évaluation du millieu stage",
+                        evaluationMillieuStageToMap(millieuStageEvaluationInfoDTO, evaluationMillieuStage)))
+                .contractId(contractId)
+                .build());
+
+    }
+
+    private Map<String, Map<String, String>> evaluationMillieuStageToMap(MillieuStageEvaluationInfoDTO millieuStageEvaluationInfoDTO,
```

Comment:
This method is very long and I don't get why you use a Map<String, Map<String, String>>

Reply:
I feel like its easier to deal with maps than to create an object that would basicly do the same job but in a more complicated way